icu4c/source/i18n/uspoof.cpp

```diff
@@ -538,6 +539,109 @@ uspoof_areConfusableUnicodeString(const USpoofChecker *sc,
     return result;
 }
 
+U_CAPI uint32_t U_EXPORT2 uspoof_areBidiConfusable(const USpoofChecker *sc, UBiDiDirection direction,
+                                                  const char16_t *id1, int32_t length1,
+                                                  const char16_t *id2, int32_t length2,
+                                                  UErrorCode *status) {
+    SpoofImpl::validateThis(sc, *status);
```

Comment:
Removed the redundant checks.
However, I do not see how
> The "if (length1 < -1 || length2 < -1) " check is contradict to the API design.

```
 * @param length1 the length of the first identifier, expressed in
 *                16 bit UTF-16 code units, or -1 if the string is
 *                nul terminated.
```
The length of the first identifier cannot be negative, so negative values other than -1 are disallowed.
Likewise, passing a length less than -1 to the UnicodeString constructor [is invalid](https://github.com/unicode-org/icu/blob/667ee72b7c2921d28b935af4e620832054452a09/icu4c/source/common/unistr.cpp#L228-L233) (though UnicodeString [signals errors using the bogus flag rather than error codes](https://github.com/unicode-org/icu/blob/667ee72b7c2921d28b935af4e620832054452a09/icu4c/source/common/unicode/unistr.h#L260-L266)).

Note that the contract is the same as in the old non-bidi version, from which I took those checks.